docs/gke.md

```diff
@@ -0,0 +1,73 @@
+# Google Kubernetes Engine (GKE) Deployment Guide
+
+Federation v2 can be deployed to and manage [GKE](https://cloud.google.com/kubernetes-engine/) clusters. Since
+Federation v2 requires Kubernetes v1.11 or greater, you must create
+[Alpha Clusters](https://cloud.google.com/kubernetes-engine/docs/concepts/alpha-clusters) and manually specify
+the `cluster-version`. Alpha clusters will no longer be needed once GKE supports Kubernetes `v1.11` or greater. You can
+use the `$ gcloud container get-server-config` command to view the GKE cluster versions available to you.
+The following example deploys two GKE alpha clusters named `cluster1` and `cluster2` using Kubernetes version
+`1.11.2-gke.4`.
+
+```bash
+$ gcloud container clusters create cluster1 --enable-kubernetes-alpha --cluster-version 1.11.2-gke.4 \
+--no-enable-autorepair --no-enable-autoupgrade
+$ gcloud container clusters create cluster2 --enable-kubernetes-alpha --cluster-version 1.11.2-gke.4 \
+--no-enable-autorepair --no-enable-autoupgrade
+```
+**NOTE:** GKE alpha clusters expire after 30 days.
+
+Configure cluster access for `kubectl`:
+```bash
+$ gcloud container clusters get-credentials cluster1
+$ gcloud container clusters get-credentials cluster2
+```
+
+If you are following along with the Federation v2 [User Guide](./userguide.md), change the cluster context names.
+First, list the context names:
+```bash
+$ kubectl config view
+```
+
+Change the generated name for each context with the cluster names used by the Federation v2 User Guide:
+```bash
+$ kubectl config $GKE_GENERATED_NAME cluster1
```

Comment:
Not familiar with GKE, where can I get this env variable `GKE_GENERATED_NAME`? Would be good if you add more detail here.

Reply:
I think that will be self-evident to anyone following this guide.

Reply:
Thanks @quinton-hoole-2 , if this is self-evident to the GKE users, +1 for @danehans to ignore this comment.

Reply:
I prefer to have these commands fully automated. This can be done by setting a variable for the `zone` and `region` the user creates the clusters in. For example:

```bash
ZONE=us-central1-f
REGION=${ZONE/-f/}
gcloud container clusters create cluster1 --zone ${ZONE} --enable-kubernetes-alpha --cluster-version 1.11.2-gke.4 \
￼ --no-enable-autorepair --no-enable-autoupgrade
gcloud container clusters create cluster2 --zone ${ZONE} --enable-kubernetes-alpha --cluster-version 1.11.2-gke.4 \
￼ --no-enable-autorepair --no-enable-autoupgrade
```

Then the rest can be automated by using the project ID:

```bash
export GCP_PROJECT=$(gcloud config list --format='value(core.project)')
kubectl config rename-context gke_${GCP_PROJECT}_${ZONE}_gke-${REGION} gke-${REGION}
```

Reply:
Renaming to `gke-${REGION}` will differ from the `cluster1/cluster2` naming scheme used by the Fed-v2 User Guide.

Reply:
Ah yes, that was a typo...I've updated the comment.

Reply:
ok, just wanted to make sure we're on the same page. I'll update the doc.